create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandlerTest.java

```diff
@@ -1,397 +1,532 @@
+                getNotFoundPublicationServiceResponse(EXTERNAL_ERROR_MESSAGE_DECORATION + PUBLICATION_IDENTIFIER
+                        + " " + EASY_TO_SEE + PUBLICATION_IDENTIFIER));
+    }
 
-        createPresignedDownloadUrlHandler.handleRequest(inputStream(IDENTIFIER_VALUE, IDENTIFIER_FILE_VALUE),
-                output, context);
+    @Test
+    void handlerReturnsServiceUnavailableResponseOnServerErrorResponseFromPublicationService() throws IOException,
+            InterruptedException {
+        var publicationService = mockUnresponsivePublicationService();
+        var handler = new CreatePresignedDownloadUrlHandler(publicationService,
+                getAwsS3ServiceReturningPresignedUrl(), mockEnvironment());
 
-        GatewayResponse<CreatePresignedDownloadUrlResponse> gatewayResponse = GatewayResponse.fromOutputStream(output);
-        assertEquals(SC_NOT_FOUND, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        handler.handleRequest(createRequest(OWNER_USER_ID, SOME_RANDOM_IDENTIFIER, FILE_IDENTIFIER), output, context);
+
+        GatewayResponse<Problem> gatewayResponse = GatewayResponse.fromOutputStream(output);
+        assertBasicRestRequirements(gatewayResponse, SC_SERVICE_UNAVAILABLE, APPLICATION_PROBLEM_JSON);
+        assertProblemEquivalence(gatewayResponse, getServiceUnavailableProblem(SOME_RANDOM_IDENTIFIER));
+    }
+
+    private void assertProblemEquivalence(GatewayResponse<Problem> gatewayResponse, Problem expected)
+            throws JsonProcessingException {
+        var actual = gatewayResponse.getBodyObject(Problem.class);
+        assertThat(actual.getStatus(), equalTo(expected.getStatus()));
+        assertThat(actual.getTitle(), equalTo(expected.getTitle()));
+        assertThat(actual.getDetail(), equalTo(expected.getDetail()));
     }
 
     @Test
-    @DisplayName("handler Returns Not Found Response On Malformed Resource Identifier")
-    public void handlerReturnsNotFoundResponseOnMalformedIdentifier() throws IOException, ApiGatewayException {
-        when(publicationService.getPublication(any(String.class)))
-            .thenThrow(new NotFoundException(ERROR_PUBLICATION_NOT_FOUND_FOR_IDENTIFIER + IDENTIFIER_VALUE));
-        createPresignedDownloadUrlHandler.handleRequest(inputStream(IDENTIFIER, IDENTIFIER_FILE_VALUE), output,
-                context);
-
-        GatewayResponse<CreatePresignedDownloadUrlResponse> gatewayResponse = GatewayResponse.fromOutputStream(output);
-        assertEquals(HttpURLConnection.HTTP_NOT_FOUND, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+    void shouldReturnNotFoundResponseOnUnknownFileIdentifier() {
+    }
```

Comment:
A bit weird that this isn't in place, but there is no check that the file actually exists…